readthedocs/api/v3/serializers.py

```diff
@@ -550,3 +560,46 @@ def get_from_url(self, obj):
     def get_to_url(self, obj):
         # Overridden only to return ``None`` when the description is ``''``
         return obj.to_url or None
+
+
+class EnvironmentVariableLinksSerializer(BaseLinksSerializer):
+    _self = serializers.SerializerMethodField()
+    project = serializers.SerializerMethodField()
+
+    def get__self(self, obj):
+        path = reverse(
+            'projects-environmentvariables-detail',
+            kwargs={
+                'parent_lookup_project__slug': obj.project.slug,
+                'environmentvariable_pk': obj.pk,
+            },
+        )
+        return self._absolute_url(path)
+
+    def get_project(self, obj):
+        path = reverse(
+            'projects-detail',
+            kwargs={
+                'project_slug': obj.project.slug,
+            },
+        )
+        return self._absolute_url(path)
+
+
+class EnvironmentVariableSerializer(serializers.ModelSerializer):
+
+    value = serializers.CharField(write_only=True)
+    project = serializers.SlugRelatedField(slug_field='slug', read_only=True)
+    _links = EnvironmentVariableLinksSerializer(source='*', read_only=True)
+
+    class Meta:
+        model = EnvironmentVariable
+        fields = [
+            'pk',
+            'created',
+            'modified',
+            'name',
+            'value',
+            'project',
```

Comment:
do we really need the project here? (haven't seen the other endpoints)

Reply:
We should just make sure it's the same everywhere, unless we have a good reason (and a code comment)

Reply:
The project is the main relation with the object(s). I think it's good to have it here. All the other responses also relates with the project.

Actually, VersionSerializer is the only one that does not returns `project`. I think I should add it there as well to keep consistency.